frontend/server/src/DAO/Contests.php

```diff
@@ -500,17 +501,24 @@ final public static function getContestsParticipating(
         $params[] = intval($offset);
         $params[] = intval($pageSize);
 
-        /** @var list<array{admission_mode: string, alias: string, contest_id: int, contestants: int, description: string, finish_time: \OmegaUp\Timestamp, last_updated: \OmegaUp\Timestamp, organizer: string, original_finish_time: \OmegaUp\Timestamp, partial_score: bool, problemset_id: int, recommended: bool, rerun_id: int, scoreboard_url: string, scoreboard_url_admin: string, start_time: \OmegaUp\Timestamp, title: string, window_length: int|null}> */
-        return \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $params);
+        /** @var list<array{admission_mode: string, alias: string, contest_id: int, contestants: int, description: string, finish_time: \OmegaUp\Timestamp, last_updated: \OmegaUp\Timestamp, organizer: string, original_finish_time: \OmegaUp\Timestamp, partial_score: bool, participating: int, problemset_id: int, recommended: bool, rerun_id: int, scoreboard_url: string, scoreboard_url_admin: string, start_time: \OmegaUp\Timestamp, title: string, window_length: int|null}> */
+        $rs = \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $params);
+
+        $contests = [];
+        foreach ($rs as $row) {
+            $row['participating'] = boolval($row['participating']);
+            $contests[] = $row;
+        }
+        return $contests;
     }
 
     /**
      * Returns all recent public contests.
      *
-     * @return list<array{admission_mode: string, alias: string, contest_id: int, contestants: int, description: string, finish_time: \OmegaUp\Timestamp, last_updated: \OmegaUp\Timestamp, organizer: string, original_finish_time: \OmegaUp\Timestamp, partial_score: bool, problemset_id: int, recommended: bool, rerun_id: int, start_time: \OmegaUp\Timestamp, title: string, window_length: int|null}>
+     * @return list<array{admission_mode: string, alias: string, contest_id: int, contestants: int, description: string, finish_time: \OmegaUp\Timestamp, last_updated: \OmegaUp\Timestamp, organizer: string, original_finish_time: \OmegaUp\Timestamp, partial_score: bool, participating: bool, problemset_id: int, recommended: bool, rerun_id: int, start_time: \OmegaUp\Timestamp, title: string, window_length: int|null}>
```

Comment:
```suggestion
     * @return list<Contestv2>
```

se puede usar el nuevo tipo en todos los lugares? así no hay necesidad de andarlo re-declarando completo cada vez.

si se queja porque hay algo mal-tipado, es un error y hay que arreglarlo.

Reply:
Corrección, serán 5 pruebas no 6.

Reply:
Que crees! Estaba checando el tema de los @return types, y ví que si son casi iguales pero la diferencia son estas 2 columnas que NO ESTÁN PRESENTES:

**`..., scoreboard_url: string, scoreboard_url_admin: string, ...`**

Podría agregarlas al **`Contestv2`** como valores **opcionales** si gustas y hacer el reemplazo del @return en las funciones correspondientes.

**Fuera de eso... las pruebas ya están listas y el código está operando correctamente :)**

Reply:
puedes mejor crear un tipo nuevo llamado `ContestWithScoreboard` que tenga esos dos campos como no-opcionales?

Reply:
Pff me confundí, en realidad lo que sucede es que el PSALM Type **Contest** ya trae esas 2 columnas como requeridas (no-opcionales), al igual que **Contestv2**, sin embargo, las funciones que no tienen un tipo en específico NO retornan esas 2 columnas, en todo caso crearía un tipo **ContestWithoutScoreboard** ¿? o qué me sugieres hacer ¿?.

Reply:
ah para eso ya existe un tipo: https://github.com/omegaup/omegaup/blob/6e469a2074f34397b9d9814e0360b4963418031a/frontend/server/src/Controllers/Contest.php#L31

en general se puede ver qué tipo le asigna la función que llama a los DAOs. hay que intentar empujar esos tipos a los DAOs para evitar esta clase de confusiones.

Reply:
Listo :)

Reply:
Con esto ya se le podría dar Merge? Requiero este código en el la `main` branch para poder jalarlo para otro PR que voy a hacer.